specs/Standards/Storage.md

```diff
@@ -0,0 +1,206 @@
+
+The result:
+```bash
+{
+  total: '2350000000000000000000',
+  available: '2350000000000000000000'
+}
+```
+
+#### Account withdraws storage deposit
+
+Alice and Bob decide to withdraw their storage deposit from the `mochi` contract.
+
+**Assumptions**
+
+- Alice's account is `alice`.
+- Bob's account is `bob`.
+- Alice and Bob's account balances are `0`.
+
+**High-level explanation**
+
+Alice issues a transaction to withdraw her deposit.
+Alice wishes to withdraw the deposit from when she paid for Bob's account, but cannot.
+Bob issues a transaction to withdraw.
+
+**Technical calls**
+
+1. Alice calls `mochi::storage_withdraw({"amount": "2350000000000000000000"})`. NEAR CLI command:
+   
+   near call mochi storage_withdraw '{"amount": "2350000000000000000000"}' --accountId alice
+
+2. She suddenly remembers she paid for Bob's account registration, and checks to see if she can withdraw that balance as well, calling `mochi::storage_balance_of({"account_id": "alice"})` which indicates she has zero balance. This is because storage withdrawal is only for the predecessor account that has signed the transaction. Alice cannot withdraw for Bob.
+
+3. Bob issues the same transaction as Alice did in step 1.
+
+    near call mochi storage_withdraw '{"amount": "2350000000000000000000"}' --accountId bob
+
+## Reference-level explanation
+
+**NOTES**:
+- All amounts, balances and allowance are limited by `U128` (max value `2**128 - 1`).
+- This storage standard uses JSON for serialization of arguments and results.
+- Amounts in arguments and results have are serialized as Base-10 strings, e.g. `"100"`. This is done to avoid JSON limitation of max integer value of `2**53`.
+- To prevent the deployed contract from being modified or deleted, it should not have any access keys on its account.
+
+**Interface**:
+
+```javascript
+// The below defines the structure that will be returned for the methods:
+// * `storage_deposit`
+// * `storage_withdraw`
+// * `storage_balance_of`
+// The `total` and `available` values are string representations of unsigned 128-bit integers.
+class AccountStorageBalance {
+   total: string;
+   available: string;
+}
+
+/************************************/
+/* CHANGE METHODS on fungible token */
+/************************************/
+// Payable method that receives an attached deposit of Ⓝ for a given account.
+// Requirements:
+// * `account_id`, if provided, must be a valid account name.
+// * The account sending the transaction must have enough Ⓝ to cover the deposit.
+// If `account_id` is omitted, the sender's account will receive the attached deposit for storage. 
+// Returns the AccountStorageBalance structure
+function storage_deposit(
+    account_id: string|null
+): string {}
+
+// Withdraws a specified amount of Ⓝ for a given account that was previously reserved for storage.
+// Requirements:
+// * `amount` must be less than or equal to the storage balance for the sending account. 
```

Comment:
Updated to: "`amount` must be less than or equal to the available storage balance for the sending account."

Given that this is a plain English description, I think "available storage balance" makes more sense, since "storage balance" is the concept being modified by "available." Feel free to unresolve this comment if you disagree.